platform-dom/src/test/java/com/softicar/platform/dom/elements/time/day/DomDayPopoverTest.java

```diff
@@ -0,0 +1,76 @@
+package com.softicar.platform.dom.elements.time.day;
+
+import com.softicar.platform.common.core.clock.CurrentClock;
+import com.softicar.platform.common.core.clock.TestClock;
+import com.softicar.platform.common.date.Day;
+import com.softicar.platform.common.testing.AbstractTest;
+import com.softicar.platform.dom.DomI18n;
+import com.softicar.platform.dom.elements.DomCell;
+import com.softicar.platform.dom.elements.button.DomButton;
+import com.softicar.platform.dom.elements.popup.modal.DomPopover;
+import com.softicar.platform.dom.elements.testing.engine.IDomTestEngine;
+import com.softicar.platform.dom.elements.testing.engine.IDomTestEngineMethods;
+import com.softicar.platform.dom.elements.testing.engine.document.DomDocumentTestEngine;
+import com.softicar.platform.dom.input.DomTextInput;
+import org.junit.Before;
+import org.junit.Test;
+
+public class DomDayPopoverTest extends AbstractTest implements IDomTestEngineMethods {
```

Comment:
I think we should add the following test cases:
- What day is highlighted in the popover if the button is clicked while the input is empty?
- What day is highlighted in the popover if the button is clicked while the input contains garbage?
- What day is highlighted in the popover if the button is clicked while the input contains a valid date?
- What happens upon ESC?
- What happens if the user selects a different year or month (and therefore enforces that the previously-selected day is no longer selectable), and then dismisses the popover?